extensions/SQLServer/SQLServer/SqlServerMemory.cs

```diff
@@ -366,92 +366,105 @@ INNER JOIN
 
     /// <inheritdoc/>
     public async Task<string> UpsertAsync(string index, MemoryRecord record, CancellationToken cancellationToken = default)
+    {
+        return await this.BatchUpsertAsync(index, new[] { record }, cancellationToken);
+    }
+
+    /// <inheritdoc/>
+    public async Task<IEnumerable<string>> BatchUpsertAsync(string index, IEnumerable<MemoryRecord> records, CancellationToken cancellationToken = default)
     {
         index = NormalizeIndexName(index);
 
         if (!(await this.DoesIndexExistsAsync(index, cancellationToken).ConfigureAwait(false)))
         {
             // Index does not exist
-            return string.Empty;
+            throw new SqlServerMemoryException($"The index '{index}' does not exist.");
```

Comment:
could you use `IndexNotFound` exception?  this exception is handled to recover from errors and automatically call `CreateIndexAsync`.

(I just noticed that `IndexNotFound` should be called `IndexNotFoundException` - feel free to fix it, or I'll do that in a separate PR later on)